frontend/src/components/App/PluginSettings/PluginSettings.tsx

```diff
@@ -284,15 +275,60 @@ export function PluginSettingsPure(props: PluginSettingsPureProps) {
 export default function PluginSettings() {
   const dispatch = useDispatch();
 
-  const pluginSettings = useTypedSelector(state => state.plugins.pluginSettings);
+  const pluginData = useTypedSelector(state => state.plugins.pluginData);
+
+  const oldLocalEnabledList = localStorage.getItem('headlampPluginSettings');
+  const localEnabledList = localStorage.getItem('enabledPluginsList');
+
+  let pluginsEnabledMap;
+
+  /**
+   * Note: this is a temporary fix to handle the migration of the old plugin settings.
+   */
+  if (oldLocalEnabledList) {
+    const oldSettings = JSON.parse(oldLocalEnabledList);
+
+    pluginsEnabledMap = oldSettings.reduce(
+      (acc: Record<string, any>, p: { name: string; isEnabled: boolean }) => {
+        acc[p.name] = !!p.isEnabled;
+        return acc;
+      },
+      {} as Record<string, any>
+    );
+
+    localStorage.setItem('enabledPluginsList', JSON.stringify(pluginsEnabledMap));
+    localStorage.removeItem('headlampPluginSettings');
+
+    dispatch(setEnablePlugin(pluginsEnabledMap));
+    dispatch(reloadPage());
+  } else {
+    /**
+     * NOTE: For compatibility with old settings, we need to check if the `localEnabledList` exists.
+     * If `localEnabledList` exists, parse it and assign it to `pluginsEnabledMap`.
+     * This indicates that previous plugin settings have been saved and can be used.
+     *
+     * If `localEnabledList` does not exist, it means the settings are not initialized
+     * and no previous plugin settings have been saved. In this case, default the plugins
+     * to being disabled to allow users to turn on their desired plugins.
+     */
+    if (localEnabledList) {
+      pluginsEnabledMap = JSON.parse(localEnabledList) as Record<string, boolean>;
+    } else {
+      pluginsEnabledMap = pluginData.reduce((acc, p) => {
+        acc[p.name] = !!p.isEnabled;
+        return acc;
+      }, {} as Record<string, boolean>);
+
+      dispatch(setEnablePlugin(pluginsEnabledMap));
+      dispatch(reloadPage());
+    }
+  }
 
   return (
     <PluginSettingsPure
-      plugins={pluginSettings}
-      onSave={plugins => {
-        dispatch(setPluginSettings(plugins));
-        dispatch(reloadPage());
-      }}
+      plugins={pluginData}
+      pluginsEnabledMap={pluginsEnabledMap}
+      onSave={() => {}}
```

Comment:
So on save we do nothing? I am confused with what this PR is supposed to fix now.